hypertrace-gradle-publish-maven-central-plugin/src/main/java/org/hypertrace/gradle/publishing/PublishMavenCentralPlugin.java

```diff
@@ -38,53 +38,63 @@ public class PublishMavenCentralPlugin implements Plugin<Project> {
   public void apply(@Nonnull Project target) {
     project = target;
     this.extension =
-      project.getExtensions().create(EXTENSION_NAME, HypertracePublishMavenCentralExtension.class);
+        project
+            .getExtensions()
+            .create(EXTENSION_NAME, HypertracePublishMavenCentralExtension.class);
     this.applyMavenPublish();
     this.applySigning();
     this.applyWithJavadocJar();
     this.applyWithSourcesJar();
     this.addPublishRepository();
     this.addPublications();
-    this.applyNexusStaging();
+    this.applyNexusPlugin();
   }
 
   private void applySigning() {
-    project.getPluginManager()
-      .apply(SigningPlugin.class);
+    project.getPluginManager().apply(SigningPlugin.class);
   }
 
-  private void applyNexusStaging() {
-    project.getRootProject().getPluginManager()
-      .apply(NexusStagingPlugin.class);
+  private void applyNexusPlugin() {
+    // FIXME - modifying root from a child is problematic and an error in gradle 7
```

Comment:
This is the heart of the problem that lead me to upgrade. The plugin as is doesn't work for gradle 7, both with this PR and without it. Gradle <7 warned and ignored when modifying root, Gradle 7+ errors. Both the old third party and new third party require application on root, so our options are:
- Make this a root plugin, and have it modify children (this is what the third party does, but ugly)
- Split and make two different root and child plugins
- Make one plugin with two "modes" for how to act when applied to root vs child. Very against this though.